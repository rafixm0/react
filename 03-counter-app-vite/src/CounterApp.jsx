import { useState } from 'react';
import PropTypes from 'prop-types';


export const CounterApp = ({value}) => {
  
  const [counter, setCounter] = useState( value );
  
  const handleAdd = () => { 
    setCounter( counter + 1 );
  }

  const handleSub = () => setCounter( counter -1 );

  const handleReset = () => setCounter( value );



  return (
    <>
        <h1>CounterApp</h1>
        <h2> { counter } </h2>

        <button onClick={ handleAdd }>+1</button>
        <button onClick={ handleSub }>-1</button>
        <button onClick={ handleReset }>Reset</button>

    </>
  )
}

CounterApp.proptoTypes = {
    value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    value: 0
}