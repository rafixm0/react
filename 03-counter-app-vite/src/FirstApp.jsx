import PropTypes from 'prop-types';


const persona = {
    nombre: 'Rafa',
    subtitulo: 'Soy el mismo OBJ',
}

export const FirstApp = ({ title, subtitle }) => {
    
  return (
    <>
        <div data-testid="test-title">{ title }</div>
        <h1>{ persona.nombre }</h1>
        <p>{ persona.subtitulo }</p>
        {/* <code>{ JSON.stringify(persona) }</code> */}
    </>
  )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired   
}

FirstApp.defaultProps = {
    title: 'No hay titulo'
}