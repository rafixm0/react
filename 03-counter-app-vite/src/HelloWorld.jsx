// Toda app en react empieza con un funcional component

// export function App(){
//     return (<h1>Hola Mundo!!!</h1>);
// }

//import React from 'react'

// export const HelloWorldApp = () => {
//   return (
//     <div><h1>Hello World App</h1></div>
//   )
// }

export const HelloWorldApp = () => <div><h1>Hello World App</h1></div>;