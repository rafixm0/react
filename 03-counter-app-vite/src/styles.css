
html, body {
    background-color: #21232A;
    color: white;
    font-family: helvetica, Arial, sans-serif;;
    font-size: 1.3rem;
    padding: 30px;
}

button {
    padding: 15px;
    font-size: 1.3rem;
    margin-right: 5px;
}