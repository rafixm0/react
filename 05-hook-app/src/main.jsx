import React from 'react'
import ReactDOM from 'react-dom/client'

import './index.css'
//import { SimpleForm } from './02-useEffect/SimpleForm'
//import { FormWithCustomHook } from './02-useEffect/FormWithCustomHook'
//import { MultipleCustomHooks } from './03-examples/MultipleCustomHooks'
//import { FocusScreen } from './04-useRef/FocusScreen'
//import { Layout } from './05-useLayoutEffect/Layout'
//import { CounterApp } from './01-useState/CounterApp'
//import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook'
//import { HooksApp } from './HooksApp'
//import { Memorize } from './06-memos/Memorize'
//import { MemoHook } from './06-memos/MemoHook'
//import { CallBackHook } from './06-memos/CallBackHook'
import { Padre } from './07-tarea-memo/Padre'

ReactDOM.createRoot(document.getElementById('root')).render(
  //<React.StrictMode>
    <Padre />
  //</React.StrictMode>,
)
