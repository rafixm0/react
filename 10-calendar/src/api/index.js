//import calendarApi from "./calendarApi";

export { default as calendarApi } from './calendarApi';