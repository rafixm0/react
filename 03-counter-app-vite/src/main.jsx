import React from 'react';
import ReactDOM from 'react-dom/client';
import './styles.css';

import { FirstApp } from './FirstApp';
import { CounterApp } from './CounterApp';


ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        <CounterApp value = { 100 } />
        {/* <FirstApp title='Hola, soy Vegeta'/> */}
    </React.StrictMode>
);
