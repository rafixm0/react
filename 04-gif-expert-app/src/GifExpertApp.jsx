import React, { useState } from 'react';
import './styles.css';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';


export const GifExpertApp = () => {

    const [categories, setCategories] = useState([ 'One Punch' ]);

    const onAddCategory = ( newCategory) => {

      if ( categories.includes(newCategory)) return;

      setCategories([newCategory, ...categories]);
      
      //categories.push(newCategory);
      //console.log(categories);
      //setCategories([newCategory, ...categories]);
      //setCategories( cat => [...cat, 'Valorant']);
    }

  return (
    <>
        <h1>GifExpertApp</h1>

        <AddCategory 
          onNewCategory={ (value) => onAddCategory(value)}
        />

      
        { 
          categories.map( (category) => (
            <GifGrid 
              key = { category } 
              category = { category } />
            ))
        }
      

     
        

    </>
  )
}
