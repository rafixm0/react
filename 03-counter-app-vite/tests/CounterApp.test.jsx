import { fireEvent, render, screen } from '@testing-library/react';
import { CounterApp } from '../src/CounterApp';

describe('Pruebas sobre el archivo CounterApp', () => {
    
    test('Debe de hacer match con el snapshoot.', () => {

        const { container } = render( <CounterApp value={100} />);

        expect( container ).toMatchSnapshot();

    })


    test('Probamos que el valor inicial del contador es 100', () => {

        render( <CounterApp value={100} />);

        expect( screen.getByText(100)).toBeTruthy();
        expect( screen.getByRole('heading', {level: 2}).innerHTML).toContain('100');

    })


    test('Probamos que incremente el botón +1', () => {

        render( <CounterApp value={10} />);
        fireEvent.click( screen.getByText('+1') );

        expect( screen.getByText('11')).toBeTruthy();

    })


    test('Probamos que incremente el botón -1', () => {

        render( <CounterApp value={10} />);
        fireEvent.click( screen.getByText('-1') );

        expect( screen.getByText('9')).toBeTruthy();

    })


    test('Probamos que funciona el botón Reset', () => {

        render( <CounterApp value={10} />);
        fireEvent.click( screen.getByText('Reset') );

        expect( screen.getByText('10')).toBeTruthy();

    })


})

