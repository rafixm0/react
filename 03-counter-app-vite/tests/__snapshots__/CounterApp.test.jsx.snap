// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el archivo CounterApp Debe de hacer match con el snapshoot. 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    100
     
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
</div>
`;
